fctesting/sandbox/Dockerfile

```diff
@@ -0,0 +1,44 @@
+ARG TMP_BUILD_DIR=/tmp/build
+
+FROM rust:1.32 as firecracker-build
+
+ARG TMP_BUILD_DIR
+ARG FIRECRACKER_SRC_DIR="/firecracker"
+ARG FIRECRACKER_BUILD_DIR="$FIRECRACKER_SRC_DIR/build"
+ARG CARGO_REGISTRY_DIR="$FIRECRACKER_BUILD_DIR/cargo_registry"
+ARG FIRECRACKER_VERSION="0.15.0"
+
+# Install system dependecies
+#
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+        musl-tools
+
+# Add musl to rust std
+RUN mkdir "$TMP_BUILD_DIR" \
+    	&& rustup install 1.32.0 \
+	&& rustup override set 1.32.0 \
+        && rustup target add x86_64-unknown-linux-musl
+
+# Install Firecracker and set binaries to TMP_BUILD_DIR
+RUN git clone https://github.com/firecracker-microvm/firecracker.git
```

Comment:
Is there any particular reason you have these as separate `RUN` statements instead of chaining them together?  In general, each directive (like `RUN`, `ENV`, etc) adds a new layer to the resulting image.  When you're building images, you generally want to reduce the total number of layers except when you want to share layers between disparate images (think a base OS or language image).  This `firecracker-build` one is internal only and you never really push it as an artifact, but introducing separate layers still has a performance impact during the build as you cause Docker's storage driver (layer storage component) to create new snapshots for each one.

I'd rewrite like this:

```Dockerfile
RUN git clone https://github.com/firecracker-microvm/firecracker.git \
    cd firecracker \
    git checkout tags/v"$FIRECRACKER_VERSION" \
    cargo build --release --features vsock --target-dir /build \
    ...
```

Reply:
Had no idea that having multiple RUN commands caused such overhead was the main reason. I've went ahead and updated it based on your feedback.